presto-main/src/main/java/com/facebook/presto/operator/aggregation/histogram/GroupedTypedHistogram.java

```diff
@@ -0,0 +1,552 @@
+ *
+ * and so on to construct the value, count pairs needed. An iterator-style function, readAllValues, exists and the caller may do whatever want with the
+ * values.
+ *
+ * </pre>
+ */
+
+public class GroupedTypedHistogram
+        implements TypedHistogram
+{
+    static final float MAX_FILL_RATIO = 0.5f;
+
+    private static final int INSTANCE_SIZE = ClassLayout.parseClass(GroupedTypedHistogram.class).instanceSize();
+    private static final int EMPTY_BUCKET = -1;
+    private static final int NULL = -1;
+    private static final ProbeType PROBE_TYPE = ProbeType.SUM_OF_SQUARE;
+    private final int bucketId;
+
+    private final Type type;
+    private final BlockBuilder values;
+    private final BucketNodeFactory bucketNodeFactory;
+    // these parallel arrays represent a node in the hash table
+    private final LongBigArray counts;
+    // need to store the value of a node's groupId
+    private final LongBigArray groupIds;
+    // array of nodePointers (index in counts, valuePositions)
+    private final IntBigArray nextPointers;
+    // per groupId, we have a pointer to the first in the list of nodes for this group
+    private final LongBigArray headPointers;
+
+    private IntBigArray buckets;
+    private int nextNodePointer;
+    private int mask;
+    private int bucketCount;
+    private int maxFill;
+    // at most one thread uses this object at one time, so this must be set to the group being operated on
+    private long currentGroupId = -1;
+    private long numberOfGroups = 1;
+    private final LongBigArray valueRawHashList = new LongBigArray();
+    private final ValueStore valueStore;
+    private final IntBigArray valuePositions;
+
+    public GroupedTypedHistogram(
+            Type type,
+            int expectedCount,
+            BlockBuilder values)
+    {
+        this.type = type;
+        this.bucketId = expectedCount;
+        this.bucketCount = computeBucketCount(expectedCount, MAX_FILL_RATIO);
+        this.mask = bucketCount - 1;
+        this.maxFill = calculateMaxFill(bucketCount, MAX_FILL_RATIO);
+        this.values = values;
+
+        checkArgument(expectedCount > 0, "expectedSize must be greater than zero");
+
+        buckets = new IntBigArray(-1);
+        buckets.ensureCapacity(bucketCount);
+        counts = new LongBigArray();
+        counts.ensureCapacity(bucketCount);
+        groupIds = new LongBigArray(-1);
+        valuePositions = new IntBigArray();
+        valueRawHashList.ensureCapacity(bucketCount);
+        nextPointers = new IntBigArray(NULL);
+        nextPointers.ensureCapacity(bucketCount);
+        // size will be set by external call, other
+        headPointers = new LongBigArray(NULL);
+
+        // index into counts/valuePositions
+        nextNodePointer = 0;
+        bucketNodeFactory = this.new BucketNodeFactory();
+        valueStore = new ValueStore(expectedCount, values);
+    }
+
+    /**
+     * @param groupId
```

Comment:
These javadocs are unnecessary (except for "block").

Reply:
so we ommit params if there's nothing to say about them and add docs only for params ?  (ie we dont' keep the lists consistent?)

just checking as I've read alternative styles

Reply:
i've changed for now, i just want to understand

Reply:
cleaned up all javadoc w/o content